src/analyzing/AST_to_IL.ml

```diff
@@ -685,6 +685,52 @@ and expr_aux env ?(void = false) e_gen =
         | _ -> exp
       in
       ident_function_call_hack exp
+  (* x = ClassName(args ...) in Python *)
+  (* ClassName has been resolved to __init__ by the pro engine. *)
+  (* Identified and treated as x = New ClassName(args ...) to support
+     field sensitivity. *)
+  | G.Assign
+      ( ({
+           e =
+             ( G.N (G.Id ((_, _), lhs_info))
+             | G.DotAccess (_, _, FN (Id ((_, _), lhs_info))) );
```

Comment:
We don't support this case yet in your `mk_class_construction` helper, for now just `x = new ...`, where `x` is a simple variable. I think it's better to have a separate PR to allow `DotAccess` and then we want to do it not only here but also for Java & co.

Reply:
I am not sure what would need to change to support it, but I will remove it. I thought I abstracted over the various callees by calling `lval` on `lhs_e` but I will trust your judgement and remove this case.